### PR TITLE
chore(engine): adapt postgres backend connection URI

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
postgres connection the URI scheme designator can be either postgresql:// or postgres://.

refer: https://www.postgresql.org/docs/10/libpq-connect.html#LIBPQ-CONNSTRING

### CODE REVIEW

**Review by dpgaspar — APPROVED** [review]

**Comment by zhaoyongjie** [comment]
I will increase the UT to prevent the coverage rate decrease. Please wait for the merge. @dpgaspar

**Review by villebro — COMMENTED** [review]
To simplify this somewhat more, we could make `BaseEngineSpec.engine: Union[str, List[str]]` and add logic for mapping all keys to the engine here if it is indeed a `List[str]`: https://github.com/apache/incubator-superset/blob/88af85ac53b62cef9b95fb2f7a069bd460d19c29/superset/db_engine_specs/__init__.py#L51

I suspect there are other engines that accept multiple engine dialect names, so this way we wouldn't need to create child classes for all synonyms.

Let me know what you think!

**Comment by zhaoyongjie** [comment]
> To simplify this somewhat more, we could make `BaseEngineSpec.engine: Union[str, List[str]]` and add logic for mapping all keys to the engine here if it is indeed a `List[str]`:
> 
> https://github.com/apache/incubator-superset/blob/88af85ac53b62cef9b95fb2f7a069bd460d19c29/superset/db_engine_specs/__init__.py#L51
> 
> I suspect there are other engines that accept multiple engine dialect names, so this way we wouldn't need to create child classes for all synonyms.
> 
> Let me know what you think!

@villebro, thanks for you remind. 
I read the related logic carefully. Using BaseEngineSpec.engine: Union[str, List[str]] may not be a good way, for the following reasons:
1. There are already many places where database.spec.engine is used to get dialect name. If we modify the engine datetype, it means that many places need to be adjusted accordingly
2. SQLAlchemy dialect name received is a string datatype. ref: https://github.com/zzzeek/sqlalchemy/blob/34f8622b687f33dadff4f8e81532b3f4050246f4/lib/sqlalchemy/engine/interfaces.py#L38

I refer to SQLAlchemy "translate" method support "postgres" and "postgresql". https://github.com/zzzeek/sqlalchemy/blob/6eda3aa8c5f58e415a49ac8dc73eb8608f4be306/lib/sqlalchemy/dialects/__init__.py#L21

**Comment by villebro** [comment]
> 1. There are already many places where database.spec.engine is used to get dialect name. If we modify the engine datetype, it means that many places need to be adjusted accordingly

I may be wrong, but the db engine specs are mostly used to get values back without needing to check for what the engine itself is. Was there some place in the code that referenced the value of `BaseEngineSpec.engine` directly, other than `db_engine_specs/__init__.py`?

> 2. SQLAlchemy dialect name received is a string datatype. ref: https://github.com/zzzeek/sqlalchemy/blob/34f8622b687f33dadff4f8e81532b3f4050246f4/lib/sqlalchemy/engine/interfaces.py#L38

This shouldn't cause trouble, as we'll be mapping all values in the list to the engines dict in https://github.com/apache/incubator-superset/blob/8676c3ebd6f49fd8102e68d682939728184b5638/superset/db_engine_specs/__init__.py#L51

> I refer to SQLAlchemy "translate" method support "postgres" and "postgresql". https://github.com/zzzeek/sqlalchemy/blob/6eda3aa8c5f58e415a49ac8dc73eb8608f4be306/lib/sqlalchemy/dialects/__init__.py#L21

Hmm, I wonder if we could leverage this to automatically populate the dialect aliases..

**Comment by zhaoyongjie** [comment]
> > 1. There are already many places where database.spec.engine is used to get dialect name. If we modify the engine datetype, it means that many places need to be adjusted accordingly
> 
> I may be wrong, but the db engine specs are mostly used to get values back without needing to check for what the engine itself is. Was there some place in the code that referenced the value of `BaseEngineSpec.engine` directly, other than `db_engine_specs/__init__.py`?

In superset.views.core have some spec.engine to get engine name, like: 
- [L1951](https://github.com/apache/incubator-superset/blob/2c649ac20f049804b1e52259add4352282d1b28c/superset/views/core.py#L1951) 
- [L2087](https://github.com/apache/incubator-superset/blob/2c649ac20f049804b1e52259add4352282d1b28c/superset/views/core.py#L2087)
- [L2091](https://github.com/apache/incubator-superset/blob/2c649ac20f049804b1e52259add4352282d1b28c/superset/views/core.py#L2091)

image


> 
> > 1. SQLAlchemy dialect name received is a string datatype. ref: https://github.com/zzzeek/sqlalchemy/blob/34f8622b687f33dadff4f8e81532b3f4050246f4/lib/sqlalchemy/engine/interfaces.py#L38
> 
> This shouldn't cause trouble, as we'll be mapping all values in the list to the engines dict in
> 
> https://github.com/apache/incubator-superset/blob/8676c3ebd6f49fd8102e68d682939728184b5638/superset/db_engine_specs/__init__.py#L51
> 
> > I refer to SQLAlchemy "translate" method support "postgres" and "postgresql". https://github.com/zzzeek/sqlalchemy/blob/6eda3aa8c5f58e415a49ac8dc73eb8608f4be306/lib/sqlalchemy/dialects/__init__.py#L21
> 
> Hmm, I wonder if we could leverage this to automatically populate the dialect aliases..

If we modify the engine datatype, these places may need to be adjusted

File: superset/db_engine_specs/__init__.py
Comment (villebro) [inline superset/db_engine_specs/__init__.py:39]: Good idea! Let's add a property `BaseEngineSpecs.aliases: List[str] = []` (I'm open to naming suggestions, perhaps 'engine_aliases' might be even more descriptive). This way we can then override this in `PostgresEngineSpec` by defining `aliases = ["postgres"]`.

File: superset/db_engine_specs/__init__.py
Comment (villebro) [inline superset/db_engine_specs/__init__.py:55]: ..Here we can then do a loop over the aliases, mapping them to the original engine class.

**Review by villebro — COMMENTED** [review]
This is converging nicely! One last suggestion for implementing the engine aliases, what do you think?

File: superset/db_engine_specs/__init__.py
Comment (zhaoyongjie) [reply superset/db_engine_specs/__init__.py:55]: Nice suggestion! I will implement this soon.

File: tests/db_engine_specs/base_tests.py
Comment (villebro) [inline tests/db_engine_specs/base_tests.py:20]: This needs to be added back, as it's needed by tests.

**Review by villebro — COMMENTED** [review]
LGTM once the import is re-added to `base_tests.py`

File: tests/db_engine_specs/base_tests.py
Comment (zhaoyongjie) [reply tests/db_engine_specs/base_tests.py:20]: ok. added back as soon

File: superset/db_engine_specs/__init__.py
Comment (villebro) [inline superset/db_engine_specs/__init__.py:56]: This can probably be simplified as follows:
```suggestion
            # populate engine alias name to engine dictionary
            for engine_alias in attribute.engine_aliases or []:
                engines[engine_alias] = attribute
```

**Review by villebro — APPROVED** [review]
LGTM! Will merge once CI passes

**Comment by zhaoyongjie** [comment]
> LGTM! Will merge once CI passes

Thank you very much for your patient

**Pull request merged**
